.github/workflows/gh-pages.yaml

```diff
@@ -48,6 +48,33 @@ jobs:
       - name: Install dependencies
         run: python3 -m pip install -r docs/requirements.txt
 
+      - name: Find and replace "%pip install cleanlab" with master branch in .ipynb files
+        if: ${{ github.ref_type == 'branch' }}
```

Comment:
will this work for when we merge PRs into master also?

Reply:
Yes, it should. It may look a little odd because there isn't an `if...else` statement here, so it's just 2 separate `if` statements - one for when the doc CI is triggered by a push to the `master` branch, another when it's triggered by a newly published release.

In pseudocode, the above-highlighted code snippet does the following:
1. If it's triggered by a push to a branch, i.e. the `master` branch in the `cleanlab/cleanlab` repo AKA our production repo
2. find `%pip install cleanlab  # for colab` in the `.ipynb` files and replace them with pip installing a specific commit hash of cleanlab. For example, if the doc build is run in the **`cleanlab/cleanlab`** repo and the merging commit hash is **`abc123`**, then the replacement will be:

`%pip install git+https://github.com/`**`cleanlab`**`/cleanlab.git@`**`abc123`**

if this is unnecessarily specific, setting it to download to a specific branch would work too, e.g. something like:

`%pip install git+https://github.com/`**`cleanlab`**`/cleanlab.git@`**`master`**